### PR TITLE
Move error log <ha-card> + color in log entries

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

1. Moving the error log to its own `<ha-card>` for consistent visual appearance.
2. Coloring in both system and error log for easy spotting.

image

image

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/logs/error-log-card.ts [hunk 1]
```diff
@@ -19,21 +19,34 @@ class ErrorLogCard extends LitElement {
...
-                @click=${this._refreshErrorLog}
-              ></ha-icon-button>
+              <ha-card>
+                <ha-icon-button
+                  icon="hass:refresh"
+                  @click=${this._refreshErrorLog}
+                ></ha-icon-button>
+                <div class="card-content error-log">
+                  ${this._errorLog.split("\n").map((entry) => {
+                    if (entry.includes("ERROR"))
```
Comment (bramkragten) [inline src/panels/config/logs/error-log-card.ts:32]: critical and fatal should also be labeled as error

File: src/panels/config/logs/error-log-card.ts [hunk 2]
```diff
@@ -19,21 +19,34 @@ class ErrorLogCard extends LitElement {
...
-                icon="hass:refresh"
-                @click=${this._refreshErrorLog}
-              ></ha-icon-button>
+              <ha-card>
+                <ha-icon-button
+                  icon="hass:refresh"
+                  @click=${this._refreshErrorLog}
+                ></ha-icon-button>
+                <div class="card-content error-log">
+                  ${this._errorLog.split("\n").map((entry) => {
```
Comment (bramkragten) [inline src/panels/config/logs/error-log-card.ts:31]: Don't do this in render, as we will then do it on every update of hass, move it to `_refreshErrorLog`

File: src/panels/config/logs/error-log-card.ts [hunk 3]
```diff
@@ -87,10 +78,30 @@ class ErrorLogCard extends LitElement {
...
   private async _refreshErrorLog(): Promise<void> {
-    this._errorLog = this.hass.localize("ui.panel.config.logs.loading_log");
+    this._errorHTML = [
+      html`${this.hass.localize("ui.panel.config.logs.loading_log")}`,
+    ];
     const log = await fetchErrorLog(this.hass!);
-    this._errorLog =
-      log || this.hass.localize("ui.panel.config.logs.no_errors");
+
+    this._errorHTML = log.split("\n").map((entry) => {
```
Comment (bramkragten) [inline src/panels/config/logs/error-log-card.ts:86]:
```suggestion
    this._errorHTML = log ? log.split("\n").map((entry) => { ... } : this.hass.localize("ui.panel.config.logs.no_errors");
```

File: src/panels/config/logs/error-log-card.ts [hunk 4]
```diff
@@ -15,29 +15,20 @@ import { HomeAssistant } from "../../../types";
 class ErrorLogCard extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
-  @internalProperty() private _errorLog?: string;
+  @internalProperty() private _errorHTML!: TemplateResult[];
```
Comment (bramkragten) [inline src/panels/config/logs/error-log-card.ts:18]:
```suggestion
  @internalProperty() private _errorHTML!: TemplateResult[] | string;
```

File: src/panels/config/logs/error-log-card.ts [hunk 5]
```diff
@@ -87,10 +78,30 @@ class ErrorLogCard extends LitElement {
...
+        entry.includes("FATAL") ||
+        entry.includes("CRITICAL")
+      )
+        return html`<div class="error">${entry}</div>`;
+
+      if (entry.includes("WARNING"))
+        return html`<div class="warning">${entry}</div>`;
+
+      return html`<div class="info">${entry}</div>`;
+    });
```
Comment (bramkragten) [inline src/panels/config/logs/error-log-card.ts:98]: Move this above the error check, as it has less checks

File: src/panels/config/logs/error-log-card.ts (hunk 4 above)
Comment (spacegaier) [reply src/panels/config/logs/error-log-card.ts:18]: Still have to get used to the fact that in TS a variable can have multiple types 🙄 ...

**Review by bramkragten — APPROVED** [review]

**Pull request merged**